File: Frontend/src/app/interfaces/product.ts
export interface Product {
    id: number;
    code: string;
    name: string;
    description: string;
    price: number;
    quantity?: number | undefined;
    pharmacy: {
      id: number;
      name: string;  
    };
  }

  export class ProductRequest {
    code: string;
    name: string;
    description: string;
    price: number;
    pharmacyName: string = "";
    quantity?: number | undefined;

    constructor(code: string, name: string, description: string,  price: number, 
       pharmacyName: string){
      this.code = code;
      this.name = name;
      this.description = description;
      this.price = price;
      this.pharmacyName = pharmacyName;
    }
  }

File: Backend/PharmaGo.Domain/Entities/Product.cs
using System.Text.RegularExpressions;
using PharmaGo.Exceptions;

namespace PharmaGo.Domain.Entities;


public class Product
{
    public int Id { get; set; }
    public string Code { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public decimal Price { get; set; }
    public Pharmacy? Pharmacy { get; set; }

    public void ValidOrFail()
    {
        if (string.IsNullOrEmpty(Code)|| Code.Length != 5 || !EsNumero(Code) || string.IsNullOrEmpty(Name)|| Name.Length > 30  || string.IsNullOrEmpty(Description) || Description.Length > 70
                || Price <= 0
                ||  Pharmacy == null)
        {
            throw new InvalidResourceException("The product is not correctly created.");
        }
    }

    static private bool EsNumero(string cadena)
    {
        // Utiliza una expresión regular para verificar si la cadena contiene solo números
        // El patrón regular ^\d+$ significa que debe comenzar (^) y terminar ($) con uno o más dígitos (\d).
        return Regex.IsMatch(cadena, @"^\d+$");
    }
}